### PR TITLE
Test and fix root symlink edge case in runfiles library

### PR DESCRIPTION
**Pull request by fmeum**
Due to incomplete test coverage for the case of a root symlink named like a Bazel module, the Bash runfiles libraries behaved incorrectly in this case. This commit fixes that and adds proper test coverage for all libraries by adding repo mappings that conflict with the existing rlocation testcase for `config.json`.

### CODE REVIEW

**Comment by fmeum** [comment]
@Wyverald

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by kshyanashree** [comment]
@bazel-io fork 6.1.0

File: tools/cpp/runfiles/runfiles_test.cc [hunk 1]
```diff
@@ -587,9 +587,11 @@ TEST_F(RunfilesTest, PathsFromEnvVars) {
 TEST_F(RunfilesTest, ManifestBasedRlocationWithRepoMapping_fromMain) {
   string uid = LINE_AS_STRING();
   unique_ptr<MockFile> rm(MockFile::Create(
-      "foo" + uid + ".repo_mapping",
-      {",my_module,_main", ",my_protobuf,protobuf~3.19.2",
-       ",my_workspace,_main", "protobuf~3.19.2,protobuf,protobuf~3.19.2"}));
+      "foo" + uid + "_repo_mapping",
```
Comment (Wyverald) [inline tools/cpp/runfiles/runfiles_test.cc:590]: how come this got changed from `foo_uid.repo_mapping` to `foo_uid_repo_mapping` and the test didn't fail?

Comment (fmeum) [reply tools/cpp/runfiles/runfiles_test.cc]: The change wasn't intentional, but the only thing that matters is that the path is the target of `_repo_mapping` in the manifest, which is guaranteed by line 598.

Comment (fmeum) [reply tools/cpp/runfiles/runfiles_test.cc]: Dropped that change to make the diff cleaner.

**Review by Wyverald — APPROVED** [review]
